pkgs/applications/misc/moz-phab/default.nix

```diff
@@ -0,0 +1,55 @@
+{ lib
+, buildPythonApplication
+, fetchPypi
+, mercurial
+# build inputs
+, distro
+, glean-sdk
+, python-hglib
+, sentry-sdk
+, setuptools
+}:
+
+buildPythonApplication rec {
+  pname = "MozPhab";
+  version = "0.1.99";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-uKoMMSp5AIvB1qTRYAh7n1+2dDLneFbssfkfTTshfcs=";
+  };
+
+  patches = [
+    # Relax python-hglib requirement
+    # https://phabricator.services.mozilla.com/D131618
+    ./D131618.diff
+  ];
```

Comment:
Please use substituteInPlace in postPatch to make further adjustments easier.

Reply:
Could you elaborate? This is a patch, and I'm using `patches` to apply it. Why does this need to be changed?
If `patches` isn't idiomatic, why having it at all?

Reply:
Patching python dependencies using patches often creates merge conflicts when pinning changes. Using substituteInPlace avoids that if the patches line does not change.